packages/admin/admin/src/EditDialog.tsx

```diff
@@ -63,17 +62,33 @@ export function useEditDialog(): [React.ComponentType<IProps>, { id?: string; mo
         };
     }, [closeDialog, openAddDialog, openEditDialog]);
 
+    const parentRouterContext = React.useContext(RouterContext);
+    const saveActionRef = React.useRef<SaveAction>();
+
     const EditDialogWithHookProps = React.useMemo(() => {
-        return (props: IProps) => {
+        return (props: EditDialogProps) => {
             return (
                 <Selection>
                     <EditDialogFormApiProvider>
-                        <EditDialogInner {...props} selection={selection} selectionApi={selectionApi} api={api} />
+                        <RouterContext.Provider
```

Comment:
The only reason for this additional RouterContext is to store the saveAction for the inner component, am I right? Isn't there a better way?

Reply:
Yes - I consider this a pretter clever mechanismn :)

Reply:
But why can't the inner component access the save action in the outer context directly?

Reply:
I moved it into EditDialogInner, no need to do this from the outer component